### ISSUE TITLE
Set java arguments from panel 

### ISSUE THREAD

**Issue by wolfiepawz** [SpaceCP/Bugs#154]
Hello.

I'd like to request the possibility to set java arguments.

**Comment by tlaundal** [comment]
Hi, 
This is actually possible, although hidden. In the config directory for the daemon that runs the servers (houston), there is a directory called servers, and in that directory there is a json config file for each server. 

Here you can add extra arguments that are passed to java, and to the server jar you are using. Each argument should be its own element in the appropriate array. 

Please do not pass memory arguments to java from here (such as `-Xms` and `-Xmx`). It is also worth noting that GC arguments are usually not required, not even for bigger servers. Less is more!

**Comment by wolfiepawz** [comment]
So I can't use:

`-XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:+CMSIncrementalPacing -XX:ParallelGCThreads=2 -XX:+AggressiveOpts`

?

**Comment by JamyDev** [comment]
You can, add each one of your argument into the `java_runtime_args` array in your servers' config, which is located at `<houstondir>/config/servers/<servername.json>`.

Please restart your daemon after this to make sure the changes get applied.

**Comment by JamyDev** [comment]
A way of setting these from the panel will be available in the future.

**Comment by wolfiepawz** [comment]
`"exec": {
        "java-runtime-args": [],
        "args": "-XX:+UseConcMarkSweepGC -XX:+UseParNewGC -XX:+CMSIncrementalPacing -XX:ParallelGCThreads=2 -XX:+AggressiveOpts",
        "java-executable": null,
        "jarfile": "KCauldron.jar"
    },
`

Is this correct use?

**Comment by JamyDev** [comment]
No, args has to be a JSON array. Plus the ones you provided are runtime arguments and thus should go in the `java-runtime-args` array.

This is the correct format:

```
"exec": {
  "java-runtime-args": ["-XX:+UseConcMarkSweepGC", "-XX:+CMSIncrementalPacing", "-XX:ParallelGCThreads=2", "-XX:+AggressiveOpts"],
  "args": [],
  "java-executable": null,
  "jarfile": "KCauldron.jar"
},
```

**Comment by wolfiepawz** [comment]
That, actually didn't work..

http://bin.jwr.website/opoyirugeg.vbs

**Comment by JamyDev** [comment]
Remove each argument one by one until it actually works, then tell me which one it was please

**Comment by wolfiepawz** [comment]
Neither, actually. Somehow, the config just went back to the old save. 
It's fine now.

**Comment by wolfiepawz** [comment]
meh, didn't mean to close it.
